test/extended/openstack/utils.go

```diff
@@ -205,7 +206,14 @@ func FindStorageClassByProvider(oc *exutil.CLI, provisioner string) *storagev1.S
 	}
 	for _, sc := range scList.Items {
 		if sc.Provisioner == provisioner {
-			return &sc
+			if seek_default {
+				val, ok := sc.GetAnnotations()["storageclass.kubernetes.io/is-default-class"]
+				if ok && val == "true" {
+					return &sc
+				} else {
+					return &sc
+				}
+			}
```

Comment:
As this stands, this will always fail with `no default storage class found` (or one of the earlier failures) unless `seek_default = true`. Is that expected? I had expected to see an else. How about something like this?

```go
if seek_default {
    val, ok := sc.GetAnnotations()["storageclass.kubernetes.io/is-default-class"]
    // if we're explicitly looking for a default, the annotation must exist
    if !ok {
        e2e.Failf("could not find annotation on storageclass")
    }
    if val == "true" {
        return &sc
    }
} else {
    // if we're not looking for a default, simply return the first result
    return &sc
}
```

Reply:
You are right @stephen. It is a typo in the indentation during copy/paste :/

I'll make use of your suggestion and test. Thank you!